fontir/src/ir.rs

```diff
@@ -1890,6 +1900,43 @@ pub struct Component {
     pub transform: Affine,
 }
 
+/// Data to inform construction of [CPAL](https://learn.microsoft.com/en-us/typography/opentype/spec/cpal#palette-table-header)
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct ColorPalettes {
+    // All palettes should have the same number of colors
+    pub palettes: Vec<Vec<Color>>,
+}
+
+impl ColorPalettes {
+    pub fn new(mut palettes: Vec<Vec<Color>>) -> Self {
+        // Trivial optimization: if there is only one palette we can deduplicate it
+        if palettes.len() == 1 {
+            palettes = vec![palettes
+                .into_iter()
+                .next()
+                .unwrap()
+                .into_iter()
+                .collect::<HashSet<_>>()
+                .into_iter()
+                .collect::<Vec<_>>()];
+        }
```

Comment:
code golf but I would prefer to write this as,

```suggestion
        if let [one_palette] = palettes.as_mut_slice() {
            one_palette.sort();
            one_palette.dedup();
        }
```